File: stress/t/opt_spill_int40.f
      PROGRAM OSP40
      INTEGER*8 A0,A1,A2,A3,A4,A5,A6,A7,A8,A9
      INTEGER*8 A10,A11,A12,A13,A14,A15,A16,A17,A18,A19
      INTEGER*8 A20,A21,A22,A23,A24,A25,A26,A27,A28,A29
      INTEGER*8 A30,A31,A32,A33,A34,A35,A36,A37,A38,A39
      INTEGER*8 ACC,I
      A0=1
      A1=1
      A2=2
      A3=3
      A4=5
      A5=8
      A6=13
      A7=21
      A8=34
      A9=55
      A10=2
      A11=4
      A12=6
      A13=8
      A14=10
      A15=12
      A16=14
      A17=16
      A18=18
      A19=20
      A20=3
      A21=6
      A22=9
      A23=12
      A24=15
      A25=18
      A26=21
      A27=24
      A28=27
      A29=30
      A30=5
      A31=10
      A32=15
      A33=20
      A34=25
      A35=30
      A36=35
      A37=40
      A38=45
      A39=50
      ACC=0
      I=0
   10 IF (I.GE.7) GOTO 20
      ACC=ACC+A0+A39+A1+A38+A2+A37+A3+A36+A4+A35
      ACC=ACC+A5+A34+A6+A33+A7+A32+A8+A31+A9+A30
      ACC=ACC+A10+A29+A11+A28+A12+A27+A13+A26+A14+A25
      ACC=ACC+A15+A24+A16+A23+A17+A22+A18+A21+A19+A20
      ACC=ACC+I*A0+A20*A21
      I=I+1
      GOTO 10
   20 CONTINUE
      WRITE(*,'(A,I0)') 'acc=',ACC
      WRITE(*,'(A,I0)') 'a0=',A0
      WRITE(*,'(A,I0)') 'a20=',A20
      WRITE(*,'(A,I0)') 'a39=',A39
      END

File: stress/t/opt_spill_int40.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t a0=1,a1=1,a2=2,a3=3,a4=5,a5=8,a6=13,a7=21,a8=34,a9=55,
    a10=2,a11=4,a12=6,a13=8,a14=10,a15=12,a16=14,a17=16,a18=18,a19=20,
    a20=3,a21=6,a22=9,a23=12,a24=15,a25=18,a26=21,a27=24,a28=27,a29=30,
    a30=5,a31=10,a32=15,a33=20,a34=25,a35=30,a36=35,a37=40,a38=45,a39=50,
    acc=0,i=0;
  for(i=0;i<7;i=i+1){
    acc=acc+a0+a39+a1+a38+a2+a37+a3+a36+a4+a35;
    acc=acc+a5+a34+a6+a33+a7+a32+a8+a31+a9+a30;
    acc=acc+a10+a29+a11+a28+a12+a27+a13+a26+a14+a25;
    acc=acc+a15+a24+a16+a23+a17+a22+a18+a21+a19+a20;
    acc=(int64_t)((uint64_t)acc+(uint64_t)((int64_t)((uint64_t)i*(uint64_t)a0))+(uint64_t)((int64_t)((uint64_t)a20*(uint64_t)a21)));
  }
  printf("acc="); printf("%ld",(long)acc); printf("\n");
  printf("a0="); printf("%ld",(long)a0); printf("\n");
  printf("a20="); printf("%ld",(long)a20); printf("\n");
  printf("a39="); printf("%ld",(long)a39); printf("\n");
  return 0;
}
